test/buy-tshirt.spec.ts

```diff
@@ -12,30 +12,52 @@ import {
-  const productAddedModalPage: ProductAddedModalPage = new ProductAddedModalPage();
-  const summaryStepPage: SummaryStepPage = new SummaryStepPage();
-  const signInStepPage: SignInStepPage = new SignInStepPage();
-  const addressStepPage: AddressStepPage = new AddressStepPage();
-  const shippingStepPage: ShippingStepPage = new ShippingStepPage();
-  const paymentStepPage: PaymentStepPage = new PaymentStepPage();
-  const bankPaymentPage: BankPaymentPage = new BankPaymentPage();
-  const orderResumePage: OrderResumePage = new OrderResumePage();
-
-  it('then should be bought a t-shirt', async () => {
+describe('Given a shopping page', () => {
+  beforeAll(async () => {
     await browser.get('http://automationpractice.com/');
-    await menuContentPage.goToTShirtMenu();
-    await productListPage.selectProduct();
-    await productAddedModalPage.proceedToCheckout();
-    await summaryStepPage.proceedToCheckout();
-    await signInStepPage.login('aperdomobo@gmail.com', 'WorkshopProtractor');
-    await addressStepPage.proceedToCheckout();
-    await shippingStepPage.acceptAndContinue();
-    await paymentStepPage.payByBankWire();
-    await bankPaymentPage.confirmOrder();
-    await expect(orderResumePage.getOrderTitle())
-      .toBe('Your order on My Store is complete.');
+  });
+
+  describe('when want to buy a T shirt', () => {
+    beforeAll(async () => {
+      const menuContentPage: MenuContentPage = new MenuContentPage();
+      const productListPage: ProductListPage = new ProductListPage();
+      const productAddedModalPage: ProductAddedModalPage = new ProductAddedModalPage();
+      const summaryStepPage: SummaryStepPage = new SummaryStepPage();
+      await menuContentPage.goToTShirtMenu();
+      await productListPage.selectProduct();
+      await productAddedModalPage.proceedToCheckout();
+      await summaryStepPage.proceedToCheckout();
+    });
+
+    describe('and login to the application', () => {
+      beforeAll(async () => {
+        const signInStepPage: SignInStepPage = new SignInStepPage();
+        await signInStepPage.login('aperdomobo@gmail.com', 'WorkshopProtractor');
+      });
+
+      describe('and select default address', () => {
+        beforeAll(async () => {
+          const addressStepPage: AddressStepPage = new AddressStepPage();
+          await addressStepPage.proceedToCheckout();
+        });
+
+        describe('and pay to the bank', () => {
+          beforeAll(async () => {
+            const shippingStepPage: ShippingStepPage = new ShippingStepPage();
+            const paymentStepPage: PaymentStepPage = new PaymentStepPage();
+            const bankPaymentPage: BankPaymentPage = new BankPaymentPage();
+            const orderResumePage: OrderResumePage = new OrderResumePage();
+            await shippingStepPage.acceptAndContinue();
+            await paymentStepPage.payByBankWire();
+            await bankPaymentPage.confirmOrder();
+
+            it('then should be bought a t-shirt', async () => {
```

Comment:
el `it` debe estar por fuera del `beforeAll` como un hijo del `describe` de la línea 45, si no no se va a ejecutar la prueba, de hecho mira la ejecución del CI que que esta prueba no se está corriendo, solo la del otro file

![image](https://user-images.githubusercontent.com/1211491/48265900-c965f980-e3fb-11e8-8ae8-9aedec66f632.png)

Reply:
entiendo... :D gracias  ya subi la correcion de nuevo